### PR TITLE
ci: parallel pytest execution

### PR DESCRIPTION
**Pull request by sylvlecl**
Experiment the use of pytest-xdist to speed up test execution in CI and dev environment:
see for example [this job](https://github.com/AntaresSimulatorTeam/AntaREST/actions/runs/10668008235/job/29566771799) where tests are executed in under 10min on ubuntu.

Following issues with parallel execution of tests are solved :
- some checks depended on the consistency at 2 moments in time of the result of `disk_usage`, which was anyway not a good idea (dependent on what happens elsewhere on the machine). It's now replaced with mocking
- some parameterized tests had random sort order, which is not allowed by pytest-xdist (all workers must have exactly the same tests in the same order)
- the `command_factory` test checked something in its teardown, assuming all tests were executed in the same process, which is not the case anymore. The same check is done as a separate unit tests.

### CODE REVIEW

File: antarest/login/auth.py
Comment (mabw-rte) [inline antarest/login/auth.py:1]: Do you have an idea why this was not corrected by `isort` ?

**Review by mabw-rte — COMMENTED** [review]
@sylvlecl  

- For the `rebase` it should be launched locally (always). 
- The `commit` messages should be corrected.
- The branch naming does not respect the `git flow` we fixed.

File: antarest/login/auth.py
Comment (sylvlecl) [reply antarest/login/auth.py:1]: Actually there is no change on import sort order in the final PR. Intermediate commits introduced unwanted changes (related to another PR).

**Comment by sylvlecl** [comment]
Closed in favor of https://github.com/AntaresSimulatorTeam/AntaREST/pull/2133 which fixes naming issues

**Pull request closed without merging**